src/dbnode/persist/fs/files.go

```diff
@@ -1126,8 +1127,31 @@ func NextIndexSnapshotFileIndex(filePathPrefix string, namespace ident.ID, block
 	return currentSnapshotIndex + 1, nil
 }
 
+// CompleteCheckpointFileExists returns whether a checkpoint file exists, and if so,
+// is it complete.
+func CompleteCheckpointFileExists(filePath string) (bool, error) {
+	f, err := os.Stat(filePath)
```

Comment:
should this have the same `strings.Contains` check?

Reply:
what would the check be? This one is specifically written to be used for checkpoint files, and if its not a checkpoint file you'll probably get an error anyways because its unlikely the file you're checking is exactly 4 bytes

Reply:
I think prateek just means, should you check that the filepath they're checking for definitely is a checkpoint file (i.e. make sure that it passes the checkpoint suffix)?  I think that makes sense.

Reply:
Ah ok, I guess I just wasn't worried about it because this function is harder to misuse but I'll add it

Reply:
done